src/pages/MyIssueDetails.js

```diff
@@ -39,10 +52,36 @@ export default function MyIssueDetails({ myIssue, avatar }) {
           <DD key={index + myIssue.id}>{label}</DD>
         ))}
       </dl>
+      <RemoveWrapper>
+        {!showRemoveDialog && (
+          <ButtonSecondary
+            children={'REMOVE'}
+            onClick={() => setShowRemoveDialog(true)}
+          />
+        )}
+        {showRemoveDialog && (
+          <RemoveDialog
+            issueId={myIssue.id}
+            setShowRemoveDialog={setShowRemoveDialog}
+            handleRemoveIssue={handleRemoveIssue}
+          />
+        )}
+      </RemoveWrapper>
     </Wrapper>
   );
+
+  function handleRemoveIssue(id) {
+    setMyIssues(myIssues.filter(myIssue => myIssue.id !== id));
+    navigate('/myissues');
+  }
```

Comment:
Since this is the only place where you need `setMyIssues`and `myIssues` in this file, I would rather write a function inside the file where the state is, and its content is the same as line 74 (setMyIssues...). Then you pass this function to `MyIssueDetail` and call it right here in line 74 with the id.

Reply:
Thx, that is cleaner code.